osu.Game/Screens/Ranking/ResultsScreen.cs

```diff
@@ -87,15 +91,20 @@ private void load()
 
             if (player != null)
             {
-                AddInternal(new HotkeyRetryOverlay
+                if (allowRetry)
```

Comment:
Maybe collapse these conditionals into one or factor out a method for adding the retry stuff? It's a bit nested right now.